workflow/introduction.rst

```diff
@@ -0,0 +1,173 @@
+
+The goal of workflow software is to streamline and track workflow activity. Since different organizations have different workflow processes, workflow software must be flexible and easy to customize.
+
+The workflow system inside of Plone is an example of a State Machine. 
+
+From `Wikipedia <http://en.wikipedia.org/wiki/Finite-state_machine>`_:
+
+.. pull-quote:: A finite-state machine (FSM) or finite-state automaton (plural: automata), or simply a state machine, is a behavioral model used to design computer programs. It is composed of a finite number of states associated to transitions. A transition is a set of actions that starts from one state and ends in another (or the same) state. A transition is started by a trigger, and a trigger can be an event or a condition.
+
+* Any object controlled by workflow is **always** in precisely one ``state`` from each workflow in its chain.
+* The ``state`` in which an object is currently located controls what ``transitions`` are available to it
+* Any workflow can be diagrammed, showing the available states and the transitions between them
+   * Diagrams like this can be of enormous help in understanding your workflow
+   * You should always sketch up a diagram when you start figuring out the workflow you want
+
+.. image:: _static/simple_workflow.png
+
+
+What's in a Workflow?
++++++++++++++++++++++
+
+Workflows control
+-----------------
+
+* What ``states`` and ``transitions`` are available
+* Which ``permissions`` will be managed (permissions not managed are left untouched from their current value by the workflow)
+* Which ``groups`` will be managed (see ``states`` below for more about this)
+* Which ``variables`` will be tracked by the workflow (values are set and stored every time a transition occurs)
+* What ``worklists`` will be generated (you can return lists of content matching values tracked by ``variables``
+* What ``scripts`` are available to be used in conjunction with ``transitions``
+
+  * These are basic python scripts, and are not used much anymore now that ``events`` are available
+
+States control
+--------------
+
+* What transitions are available **out**
+* What ``permissions`` are assigned to which ``roles`` *locally* to the object
+* What ``groups`` are assigned to which ``roles`` *locally* to the object
+
+  * This is probably the least-used aspect of workflow
+  * It can be spectacularly useful
+
+Transitions control
+-------------------
+
+* What ``state`` they will end in
+* What conditions are required for it to be available
+
+  * These can be ``permissions`` of the user, ``roles`` a user has, ``groups`` to which the user belongs, or even the boolean value of *'TALES*' expressions
```

Comment:
I suggest mentioning the wird `guards`: 

* These ``guards`` can be ``permissions`` of the user, ``roles`` a user has, ``groups`` to which the user belongs, or even the boolean value of *'TALES*' expressions. Since these expressions can refer to browser-views you can control if a transition is available by any code you like.